com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/persistence/query/Query.java

```diff
@@ -3,14 +3,21 @@
 import java.util.List;
 
 import javax.persistence.NoResultException;
+import javax.persistence.NonUniqueResultException;
 
 public interface Query<R, T extends Query<?, ?>> {
 
     T limitOnSelect(int limit);
 
     T offsetOnSelect(int offset);
 
-    R singleResult() throws NoResultException;
+    /**
+     * @return the result
+     * @throws NoResultException Thrown when there is no result. Do not use singleResult() method in case you can expect no result
+     * @throws NonUniqueResultException Thrown when there is multiple results. If you would like to get single result only, you can use
```

Comment:
Change to `Thrown when there are multiple results. If you want to get the first of these, use {@code limitOnSelect(1).list()} instead.`